fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/repository/SocialMediaRepositoryTest.java

```diff
@@ -0,0 +1,128 @@
+package com.github.dedis.popstellar.repository;
+
+import com.github.dedis.popstellar.model.objects.Chirp;
+import com.github.dedis.popstellar.model.objects.Lao;
+import com.github.dedis.popstellar.model.objects.security.MessageID;
+import com.github.dedis.popstellar.utility.error.UnknownChirpException;
+
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import io.reactivex.observers.TestObserver;
+
+import static com.github.dedis.popstellar.testutils.Base64DataUtils.generateMessageID;
+import static com.github.dedis.popstellar.testutils.Base64DataUtils.generatePublicKey;
+import static java.util.Collections.*;
+import static org.junit.Assert.*;
+
+public class SocialMediaRepositoryTest {
+
+  private static final String LAO_ID = Lao.generateLaoId(generatePublicKey(), 1000, "LAO");
+
+  private static final Chirp CHIRP_1 =
+      new Chirp(
+          generateMessageID(), generatePublicKey(), "This is a chirp !", 1001, new MessageID(""));
+  private static final Chirp CHIRP_2 =
+      new Chirp(
+          generateMessageID(),
+          generatePublicKey(),
+          "This is another chirp !",
+          1003,
+          new MessageID(""));
+
+  @Test
+  public void addingAChirpAfterSubscriptionUpdatesIds() {
+    SocialMediaRepository repo = new SocialMediaRepository();
+    TestObserver<Set<MessageID>> ids = repo.getChirpsOfLao(LAO_ID).test();
+    // make sure the current last element is an empty set
+    ids.assertValueAt(ids.valueCount() - 1, emptySet());
+    repo.addChirp(LAO_ID, CHIRP_1);
+    // make sure we received a new value : the set containing the chirp
+    ids.assertValueAt(ids.valueCount() - 1, singleton(CHIRP_1.getId()));
+  }
+
+  @Test
+  public void addingChirpBeforeSubscriptionUpdateIds() {
+    SocialMediaRepository repo = new SocialMediaRepository();
+    repo.addChirp(LAO_ID, CHIRP_1);
+    TestObserver<Set<MessageID>> ids = repo.getChirpsOfLao(LAO_ID).test();
+    // The value at subscription contains the first chirp's id
+    ids.assertValueAt(ids.valueCount() - 1, singleton(CHIRP_1.getId()));
```

Comment:
what does `singleton()` do?  do we need it ?

Reply:
Not that much, it is a usual way to create a set containing one element.

For readability, I could use the function `setOf()` yes